### PR TITLE
Gas fee optimization

### PR DESCRIPTION
**Pull request by Yingjie-Zhang-dev**
Implement gas fee optimization. 
Both files have been tested to work properly.

### CODE REVIEW

File: CauseContract.sol [hunk 1]
```diff
@@ -34,7 +37,7 @@ contract CauseContract {
     // donor proportion tracking
     mapping(address => uint256) public donorTotals;
     
-    mapping(address => bool) public addressDonated;
+    // mapping(address => bool) public addressDonated;
```
Comment (finnformica) [inline CauseContract.sol]: what is this comment for?

File: CauseContract.sol [hunk 2]
```diff
@@ -67,9 +70,14 @@ contract CauseContract {
 
     constructor(string memory _id, string memory _name, address payable _admin, string memory _description, string memory _websiteURL, string memory _thumbnailURL, string memory _email) {
         admin = _admin;
-        contractAddress = payable(address(this));
+
+        //  directly use address(this) whenever needed can save some storage and gas
+        // contractAddress = payable(address(this));
```
Comment (finnformica) [inline CauseContract.sol]: delete these comments pls

Comment (Yingjie-Zhang-dev) [reply CauseContract.sol]: No problem.

File: CauseContract.sol (hunk 1 above)
Comment (Yingjie-Zhang-dev) [reply CauseContract.sol]: I kept the original code in the form of comment for comparison with the changed code.
Here I changed mapping to array for gas saving. Calm, I just don't want to cause any confusion

**Review by finnformica — DISMISSED** [review]

File: CauseFactory.sol [hunk 3]
```diff
@@ -33,7 +33,8 @@ contract CauseFactory {
         return infos;
     }
 
-    function cfRetrieveIds() public view returns (string[] memory) {
```
Comment (finnformica) [inline CauseFactory.sol]: Could you leave this function in? I've taken a look at the docs and apprently you can only access one value of an array or mapping at a time, so returning the entire array is more efficient...

Comment (Yingjie-Zhang-dev) [reply CauseFactory.sol]: Okay🫡

Comment (finnformica) [reply CauseFactory.sol]: Could you uncomment the code pls and then I'll approve the commit

Comment (Yingjie-Zhang-dev) [reply CauseFactory.sol]: Sorry! I thought I had uploaded the ucommented file 2 days ago, but it seems like the file didn't successfully uploaded at that time. 
It should works now!

**Review by finnformica — APPROVED** [review]

**Pull request merged**